### ISSUE TITLE
Quick Reblog/Queue/Draft failing

### ISSUE THREAD

**Issue by MojoShoujo** [AprilSylph/XKit-Rewritten#206]
- The browser I'm using is: Firefox for Windows
- My browser version is: 91.0.1
- My XKit Rewritten version is: 0.6.0

---
When quick-reblogging or queueing a post, the first attempt will sometimes display a 'failed to authorize' message. If attempted again, the quick post options will stick as 'processing' for all posts until the page is hard refreshed. Posts can still be reblogged normally if stuck this way.
The amount of posts before this occurs varies. Yesterday I would be lucky if one or two would quick post before it stuck, but today I queued about a dozen before seeing the issue again.

**Comment by AprilSylph** [comment]
I've never encountered this message, and I don't hard-refresh all that often on my main machine, so I assume that if I were able to reproduce this, I would have done so accidentally already.

Have you tried clearing your Tumblr cookies? The problem might be with your login session.

**Comment by Sccherbatsky** [comment]
I made an account just to comment here, hopefully I'm able to adequately describe things despite being technically dumb. But I wanted to corroborate the issue reported by MojoShoujo. The bug described is almost identical to one I've been dealing with as for roughly over a week now. The only difference I've encountered from what was described is that I don't always have to hard refresh the page. When quick-reblogging, queuing, or drafting, if I encounter the "failed to authorize" message when trying to do something, I'm able to temporarily fix the problem by "closing" the little quick post thing (meaning, click out, scroll away, etc., just to make it disappear and close) and then reopening it. I've never received the error message on the same post again when doing this. However, I do encounter the error message frequently when scrolling my dash and posting/queuing/drafting - I'd say approximately 50% of my post interactions start with this error message. If I attempt to post/queue/draft a post _without_ "closing" the little quick post thing first, if I just click again immediately after seeing the failure to authorize message, the whole thing sticks just as MojoShoujo described until the page is hard refreshed.

Clearing my Tumblr cookies hasn't seemed to help the situation.

My information is nearly the same as MojoShoujo:

- Using Firefox for Windows version 91.0.1
- XKit Rewritten version is: 0.6.1 (Error has been occurring since 0.5.0, I believe)

**Comment by AprilSylph** [comment]
hmm...

just to be sure, does this issue persist when XKit Rewritten is the *only* addon you have enabled? I recommend restarting your browser after disabling addons, in order to be certain they're completely inactive.

**Comment by Sccherbatsky** [comment]
Nope, it doesn't. I created a new fresh profile and logged in, installed XKit Rewritten, and was able to draft like 50 posts without a single error message. That also fixed _another_ issue I've been having lately where none of my header notifications have been updating. In my main profile, the header activity button has been stuck on showing me 91 new activity notifications for over a week as well, it doesn't clear, doesn't update. I get no new post notifications, no messaging notifications, no inbox message notifications, nothing. In this fresh profile that was working as well.

So I closed that browser window and fresh profile, and disabled all of my add-ons except XKit Rewritten in my main profile, deleted all cached data and cookies again, and then closed and reopened the browser. Both the original issue and the notification persist. The first post I tried to draft gave me the "failed to authorize" message. Seems perplexing!

**Comment by AprilSylph** [comment]
Weird! I can't really guess what part of your current profile could be breaking so much - I've got a ton of `about:config` tweaks set (mostly from [PrivacyTools](https://privacytools.io/browsers/#about_config)) and while it breaks some parts of the blog network (blog controls can't resize themselves, askboxes/submissions are broken), none of it affects Tumblr's main web interface.

I'm not sure what to recommend except transferring everything over to that new profile.

**Comment by Sccherbatsky** [comment]
So the weird thing is nothing breaks, even after adding everything. In fact, if I just open a private browser from my main profile, with all of the add-ons I have still enabled, everything works. Quick everything, my header notifications, all of it. It's only when in non-private browsing with this profile that the issues continue. Maybe I just need to entirely migrate permanently to a new profile.

It's weird that someone starting encountering at least half the issue that I'm having at the same time though, as if it's not just my profile that something is wonky with.

**Comment by MojoShoujo** [comment]
It's been a busy few days but I've slowly been tinkering with settings and finding the same as Sccherbatsky. Clearing cache/history/etc doesn't fix it, and neither does turning off all my extensions (or even parts of them- eg allowing a provisionally blocked potential tracker on privacypossum, refreshing and trying again). Moving to a new profile or even private mode on the same profile fixes it. *shrug* I'm baffled

**Comment by AprilSylph** [comment]
Unfortunately there's nothing I can really do about browser profiles randomly breaking things. I'll leave this issue open in case anyone else has the same issue, and in hope that someone affected is capable of contributing a fix.

**Comment by AprilSylph** [comment]
@Sccherbatsky reading back over your comment where you "deleted all cached data and cookies again"... did you also clear "Site Preferences" and "Offline web site data"? If not, and assuming you still have the original profile with the described issue, it would be interesting if clearing those fixes the issue. I'm not sure exactly what the two options refer to, but hopefully one of them clears Tumblr's `localStorage`, which could be the issue if cookies weren't.

**Comment by Sccherbatsky** [comment]
You know what, I don't think I did that. I can't be 100% but I don't remember making sure that was checked. However, unfortunately, I deleted the profile about a week ago because I'm fully migrated to a new profile and I couldn't think of anything else to try... but something so simple like that, turns out I probably missed it, lol. So that could have helped! I'm not sure if Mojo is still following this thread, but if they still have their problem profile I'd be super interested as well to know whether this solves the problem.

The good news is that everything has been smooth sailing since I migrated, not a hiccup to speak of with anything.

**Comment by Sccherbatsky** [comment]
Well, the good news is I can now confirm that making sure that the box is checked to clear offline website data doesn't help the issue. The bad news is the whole problem has started again. Same failed to authorize bug with the quick reblog tool through xkit, but the same issue persists again, even when I disable xkit (and pallete), where my notifications are hanging on the dashboard.

I know everything was working this morning before I started work. But then when I first got on this evening after work, it was all like this. I have no idea what could have happened to my profile in that time... but I guess I need to migrate again.

**Comment by AprilSylph** [comment]
> If attempted again, the quick post options will stick as 'processing' for all posts until the page is hard refreshed.

Might be worth mentioning that this was fixed in b91b8bbf635a296a9493236a86156ec060a168c3 (v0.11.0)

**Comment by threerings13** [comment]
I have been having this problem for weeks.  Any attempt to reblog a post, add to queue, etc through Quick Reblog results in an "Unable to Authorize" error.  However, it works if I click the same thing a second time.  Sometimes a third.  Basically I have to do everything twice to make it register.  I also have the Number of new posts hanging at the top of my dashboard.  I'm on Windows 10 and Firefox.  

If I disable Xkit Rewritten entirely I don't have this issue, and I have been working on tracking it down and it's just Quick Reblog that causes the issue.  If I keep everything else in Xkit but disable Quick Reblog I don't get the "Unable to Authorize" message anymore, but the number of posts is still stuck at 99+ at the top of my dash.

This started for me a few weeks ago.

**Comment by JayWillmann** [comment]
I found a different solution for this, using "ctrl + shift + r" on tumblr to force a hard refresh totally solved it even without deleting my cache. I saw someone else report it as a firefox issue even without xkit.

**Comment by AprilSylph** [comment]
Huh, isn't <kbd>CTRL</kbd>+<kbd>SHIFT</kbd>+<kbd>R</kbd> just a cacheless reload? I wonder what differs between that and actually deleting cache via Firefox settings...

**Comment by marcustyphoon** [comment]
I certainly wouldn't be surprised if there were multiple causes for this. We should probably suggest a hierarchy of steps, with "make a new firefox profile" at the end, unless we can determine any details about what's actually happening.

**Comment by AprilSylph** [comment]
> I certainly wouldn't be surprised if there were multiple causes for this. We should probably suggest a hierarchy of steps, with "make a new firefox profile" at the end, unless we can determine any details about what's actually happening.

* Use <kbd>CTRL</kbd>+<kbd>SHIFT</kbd>+<kbd>R</kbd> to reload the page without cache
* Clear your cache for `www.tumblr.com` via `about:preferences#privacy`
* Clear your cache and cookies for `www.tumblr.com` via `about:preferences#privacy`
* Clear your cache for all sites via `about:preferences#privacy`
* Disable literally every other browser extension via `about:addons`
* Clear your cache and cookies for all sites via `about:preferences#privacy`
* Create a completely new profile via `about:profiles`

Am I missing any big ones here? Other than the unspoken "quit and re-open Firefox and then see if the problem is fixed after doing this" to be appended to each potential fix :P
